crates/services/txpool/src/txpool.rs

```diff
@@ -354,6 +369,21 @@ where
             return Err(Error::NotInsertedTxKnown)
         }
 
+        // If upgrading transition function, at least check that it is valid wasm
+        if let PoolTransaction::Upgrade(upgrade, _) = tx.as_ref() {
+            if let UpgradePurpose::StateTransition { root } =
+                upgrade.transaction().upgrade_purpose()
+            {
+                if let Err(err) = self.wasm_checker.uploaded_wasm_is_valid(root) {
+                    return Err(match err {
+                        WasmValidityError::NotEnabled => Error::NotInsertedWasmNotEnabled,
+                        WasmValidityError::NotFound => Error::NotInsertedWasmNotFound,
+                        WasmValidityError::NotValid => Error::NotInsertedInvalidWasm,
+                    });
+                }
```

Comment:
Let's create a todo to impl `From<WasmValidityError>` and create an issue to move `TxPoolError` from `fuel-core-types` to `fuel-core-txpool` crate.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2082

Reply:
45eb7fd3a

Reply:
Picking this up. Is the long term idea to move everything from `src/services/txpool.rs` in `fuel-core-types` to `src/types.rs` in `fuel-core-txpool` or is it just the error type for now?